client/src/process.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable no-undef */
+const baseURL = 'http://localhost:10000';
+
+/**
+ * Function displays the overlay
+ */
+
+export function showOverlay() {
+    document.getElementById('overlay').style.display = 'block';
+}
+
+/**
+ * Function hides the overlay
+ */
+
+export function hideOverlay() {
+    document.getElementById('overlay').style.display = 'none';
+}
+
+/**
+ * Function displays the loading spinner and overlay
+ */
+
+export function showSpinner() {
+    showOverlay();
+    document.getElementById('loadingSpinner').style.display = 'block';
+}
+
+/**
+ * Function hides the loading spinner and overlay
+ */
+
+export function hideSpinner() {
+    hideOverlay();
+    document.getElementById('loadingSpinner').style.display = 'none';
+}
+
+/**
+ * Function is called when the user clicks on the "Update" button
+ * Updates the network graph with the user ID entered in the input box
+ */
+
+export function updateID() {
+    console.log('Update ID called');
+    showSpinner();
+
+    let userID = document.getElementById('idInput').value;
+
+    if (!/^\d+$/.test(userID)) {
+        alert('Please enter a valid number.');
+        console.log('Invalid user ID entered');
+        hideSpinner();
+        return;
+    }
+
+    console.log('User ID', userID);
+
+    let apiURL = `${baseURL}/api/v1/account/${userID}/initialize`;
+
+    // Need to write a common http request function with error handling
+    try {
+        fetch(apiURL)
+            .then((response) => response.json())
+            .then((json) => {
+                try {
+                    console.log(json);
+                    const [nodes, edges] = processData(json, userID);
+                    console.log('Nodes and edges', nodes, edges);
+                    drawFromResponse(nodes, edges);
+                    hideSpinner();
+                } catch (error) {
+                    console.log('Error in processing response', error);
+                    // Handle or rethrow the error as needed
+                }
+            });
+    } catch (error) {
+        alert('Error fetching data from API. Please try again later.');
+        console.log('Error', error);
+        hideSpinner();
+    }
+}
+
+/**
+ *
+ * @param {JSON} json Response json from the API
+ * @param {Number} mainID
+ * @returns {Array} [nodes,edges]
+ *
+ * Function takes in the JSON response from the API and the main user ID
+ * Returns an array of nodes and edges to be used in the network graph
+ *
+ *
+ */
+
+export function processData(json, mainID = '109252111498807689') {
```

Comment:
If this mainID default value is intentional, pls put a comment explaining what it means.

Reply:
Nice catch - let's get rid of this default value and instead try to handle the case (log or show error) when id is not provided

Reply:
Good comment, resolving it

Reply:
Possibly use the userID verify function mentioned in the previous comment to handle case of id not valid? May be unecessary if we can guarantee that the userID being passed is already verified, but probably safer to just handle it anyway.

Reply:
I think at this point, it would already be verified. Not sure if it would provide any value having it again